src/hugr/validate.rs

```diff
@@ -72,6 +77,7 @@ impl<'a, 'b> ValidationContext<'a, 'b> {
         Self {
             hugr,
             dominators: HashMap::new(),
+            #[cfg(feature = "extension_inference")]
```

Comment:
Alternatively perhaps leave this ExtensionValidator here but make its methods not do anything?

Reply:
this is nicer 👍